domain/access/state/permission.go

```diff
@@ -432,6 +433,69 @@ AND     u.removed = false
 	return userAccess, nil
 }
 
+// UserModelAccessInfo gets info about all the models a particular user has access to.
+func (st *PermissionState) UserModelAccessInfo(ctx context.Context, name string) ([]access.UserModelAccessInfo, error) {
+	db, err := st.DB()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	getModelAccessInfo := `
+SELECT (m.name, m.uuid, t.type, mll.time) AS (&dbModelAccessInfo.*),
+       owner.name AS &dbModelAccessInfo.owner_name
+FROM   v_user_auth u
+       JOIN v_permission p ON u.uuid = p.grant_to
+       JOIN model m ON m.uuid = p.grant_on
+       JOIN user owner ON m.owner_uuid = owner.uuid
```

Comment:
You don't need this - it's contained in `v_user_auth`.  So `u.name AS &dbModelAccessInfo.owner_name`

Reply:
But what about if the owner is not the same as the user who the access is granted to? I would have thought selecting u.name as the owner would set the user who had authorization on the model as the owner

Reply:
Though I will actually have to delete this function anyway, since this has been separately implemented by Tom.